poe_set_port.go

```diff
@@ -0,0 +1,296 @@
+	if !exists {
+		return "", errors.New("could not find hash.")
+	}
+	return hash, err
+}
+
+func findMaxPwrLimitInHtml(reader io.Reader) (string, error) {
+	doc, err := goquery.NewDocumentFromReader(reader)
+	if err != nil {
+		return "", err
+	}
+
+	limit, exists := doc.Find("input.pwrLimit").Attr("value")
+	if !exists {
+		return "", errors.New("could not find power limit.")
+	}
+	return limit, err
+}
+
+func compareSettings(name string, defaultValue string, newValue string, poeExt *PoeExt) (string, error) {
+	if len(newValue) == 0 {
+		return defaultValue, nil
+	}
+
+	switch name {
+	case "PortPrio":
+		portPrio := asNumPortPrio(newValue)
+		if portPrio == "unknown" {
+			return portPrio, errors.New("port priority could not be set. Accepted values are: [low | high | critical]")
+		}
+		return portPrio, nil
+	case "PwrMode":
+		pwrMode := asNumPwrMode(newValue)
+		if pwrMode == "unknown" {
+			return pwrMode, errors.New("power mode could not be set. Accepted values are: [802.3af | legacy | pre-802.3at | 802.3at]")
+		}
+		return pwrMode, nil
+	case "LimitType":
+		limitType := asNumLimitType(newValue)
+		if limitType == "unknown" {
+			return limitType, errors.New("limit type could not be set. Accepted values are: [none | class | user]")
+		}
+		return limitType, nil
+	case "PwrLimit":
+		if defaultValue != newValue {
+			value, err := strconv.Atoi(strings.Replace(newValue, ".", "", -1))
+			if err != nil {
+				return defaultValue, errors.New("unable to check power limit")
+			}
+
+			limit, err := strconv.Atoi(strings.Replace(poeExt.PortMaxPower, ".", "", -1))
+			if err != nil {
+				return defaultValue, errors.New("unable to check power limit")
+			}
+
+			if value > limit || value < 30 {
+				return defaultValue, errors.New(fmt.Sprintf("provided power limit (W) is out of range. Minimum: %s <> Maximum: %s", "3.0", poeExt.PortMaxPower))
+			}
+
+			return newValue, nil
+		}
+		return defaultValue, nil
+	case "DetecType":
+		detecType := asNumDetecType(newValue)
+		if detecType == "unknown" {
+			return detecType, errors.New("detection type could not be set. Accepted values are: [IEEE 802 | 4pt 802.3af + Legacy | Legacy]")
+		}
+		return detecType, nil
+	default:
+		return defaultValue, errors.New("could not find port setting.")
+	}
+
+}
+
+func asNumPortPower(portPwr string) string {
+	if portPwr == "enabled" || portPwr == "enable" {
+		return "1"
+	} else if portPwr == "disabled" || portPwr == "disable" {
+		return "0"
+	} else {
+		return "unknown"
+	}
+}
+
+func asNumPwrMode(pwrMode string) string {
+	switch pwrMode {
+	case "802.3af":
+		return "0"
+	case "legacy":
+		return "1"
+	case "pre-802.3at":
+		return "2"
+	case "802.3at":
+		return "3"
+	default:
+		return "unknown"
+	}
+}
+
+func asNumPortPrio(portPrio string) string {
+	switch portPrio {
+	case "low":
+		return "0"
+	case "high":
+		return "2"
+	case "critical":
+		return "3"
+	default:
+		return "unknown"
+	}
+}
+
+func asNumLimitType(limitType string) string {
```

Comment:
I did consider the duplication here but was not quite sure how to resolve it -- since the lookup needs to be two way (converting from display values to native values for the switch), would such a mapping make sense? Or should there be separate methods of searching by key or value?

	var limitTypes = map[string]string{
		"none":  0,
		"class": 1,
		"user":  2,
		"0": "none",
		"1": "class",
		"2": "user"
	}

`map[string]string{}` should be OK here, I believe, since url.Values{} takes string types

Reply:
Good question ... I rather prefer simple mapping.
this makes it trivial in one direction, and for the reverse option, a search by key value would be logical.
Since we just have a few keys ... performance is not an issue.
I think, even if there's just one map, it's totally fine to isolate the mapping via functions - I mean in both directions.


```go
var limitTypes = map[string]int{
	"none":  0,
	"class": 1,
	"user":  2,
}
```

Reply:
fine for you?

Reply:
Indeed, thanks for the guidance! This change should greatly help with future maintainability as well 👍

I've changed over to map lookups and relocated these into `maps.go` so that it is easier to find in the future for contributors, as the values are now shared by multiple files/functions.